Compile bug: spawn.h not found (Android - Temux)

### Git commit

eadc4184caee5b5f68f31f19a2f65c6961748e46

Build b7631

### Operating systems

Other? (Please let us know in description)

### GGML backends

CPU

### Problem description & steps to reproduce

I am trying to compile the latest release of llama.cpp for Android using Termux following the Termux build steps, almost before the build completes, it generates an error stating spawn.h was not found.

The build-essential, git, gcc & cmake packages are installed on Termux.

Running Android 16 with the latest build of Termux.

### First Bad Commit

The last time I built llama.cpp on Android was b6811 so a while ago.

### Compile command

```shell
cmake -B build
cmake --build build --config Release
```

### Relevant log output

```shell
In file included from /data/data/com.termux/files/home/new/llama.cpp-b7631/tools/server/server-models.cpp:8:    /data/data/com.termux/files/home/new/llama.cpp-b7631/common/../vendor/sheredom/subprocess.h:234:10: fatal error:                                                              'spawn.h' file not found                            234 | #include <spawn.h>                                    |          ^~~~~~~~~                              1 error generated.                                      make[2]: *** [tools/server/CMakeFiles/llama-server.dir/build.make:115: tools/server/CMakeFiles/llama-server.dir/server-models.cpp.o] Error 1                            make[1]: *** [CMakeFiles/Makefile2:5124: tools/server/CMakeFiles/llama-server.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

--- a/docs/android.md
+++ b/docs/android.md
@@ -29,7 +29,7 @@ With Termux, you can install and run `llama.cpp` as if the environment were Linu
 
 ```
 $ apt update && apt upgrade -y
-$ apt install git cmake
+$ apt install git cmake libandroid-spawn
 ```
 
 Then, follow the [build instructions](https://github.com/ggml-org/llama.cpp/blob/master/docs/build.md), specifically for CMake.

